src/API/NoteActions.php

```diff
@@ -107,6 +107,11 @@ public function trigger_note_action( $request ) {
 			$note->set_status( $triggered_action->status );
 		}
 
+		// Update note state depending on its type.
+		if ( 'survey' === $note->get_type() ) {
```

Comment:
Could we use the const from WC_Admin_Note?

Reply:
Good idea! It was done in the commit 92839f6